addons/website_sale/static/src/interactions/sale_variant_mixin.js

```diff
@@ -0,0 +1,619 @@
+            default_price.parentElement.classList.add("d-none");
+            compare_price.classList.remove("d-none");
+        }
+
+        // update images & tags only when changing product
+        // or when either ids are "false", meaning dynamic products.
+        // Dynamic products don't have images BUT they may have invalid
+        // combinations that need to disable the image.
+        if (!combination.no_product_change) {
+            this.updateProductImage(
+                parentEl.closest("tr.js_product, .oe_website_sale"),
+                combination.display_image,
+                combination.product_id,
+                combination.product_template_id,
+                combination.carousel,
+                isCombinationPossible
+            );
+            parentEl.querySelector(".o_product_tags").innerHTML = combination.product_tags;
+        }
+
+        parentEl.querySelector(".product_id").value = (combination.product_id || 0)
+        parentEl.querySelector(".product_id").dispatchEvent(new Event("change"));
+
+        this.handleCustomValues(ev.target);
+    }
+
+    /**
+     * returns the formatted price
+     *
+     * @param {float} price
+     */
+    priceToString(price) {
+        var precision = 2;
+
+        if (document.querySelectorAll(".decimal_precision")) {
+            precision = parseInt(document.querySelector(".decimal_precision").last.dataset.precision);
+        }
+        const formatted = price.toFixed(precision).split(".");
+        const { thousandsSep, decimalPoint, grouping } = localization;
+        formatted[0] = insertThousandsSep(formatted[0], thousandsSep, grouping);
+        return formatted.join(decimalPoint);
+    }
+
+    /**
+     * Returns a throttled `getCombinationInfo` with a leading and a trailing
+     * call, which is memoized per `uniqueId`, and for which previous results
+     * are dropped.
+     *
+     * The uniqueId is needed because on the configurator modal there might be
+     * multiple elements triggering the rpc at the same time, and we need each
+     * individual product rpc to be executed, but only once per individual
+     * product.
+     *
+     * The leading execution is to keep good reactivity on the first call, for
+     * a better user experience. The trailing is because ultimately only the
+     * information about the last selected combination is useful. All
+     * intermediary rpc can be ignored and are therefore best not done at all.
+     *
+     * The keepLast is to make sure we only consider the result of the last call, when several
+     * (asynchronous) calls are done in parallel.
+     *
+     * @param {string} uniqueId
+     * @returns {function}
+    */
+    throttledGetCombinationInfo = memoize(function (self, uniqueId) {
+        const keepLast = new KeepLast();
+        return (ev, params) => keepLast.add(this.throttleForAnimation(this.getCombinationInfo(ev, params)));
+    })
```

Comment:
What about adding `lastKept` as a tool in `Interaction` ? or a version that combines it with throttled ?